patches/net/minecraft/world/entity/npc/VillagerTrades.java.patch

```diff
@@ -0,0 +1,22 @@
+--- a/net/minecraft/world/entity/npc/VillagerTrades.java
++++ b/net/minecraft/world/entity/npc/VillagerTrades.java
+@@ -1484,9 +_,6 @@
+         private final int villagerXp;
+ 
+         public EmeraldsForVillagerTypeItem(int p_35669_, int p_35670_, int p_35671_, Map<VillagerType, Item> p_35672_) {
+-            BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_)).findAny().ifPresent(p_339515_ -> {
```

Comment:
Instead of removing the code, add an `if (false) // Neo: disable this check so that mods can add custom villager types` above.